R/learner_xgboost_surv_xgboost_aft.R

```diff
@@ -0,0 +1,195 @@
+#' @title Extreme Gradient Boosting AFT Survival Learner
+#' @author bblodfon
+#' @name mlr_learners_surv.xgboost.aft
+#'
+#' @description
+#' eXtreme Gradient Boosting regression using an **Accelerated Failure Time**
+#' objective.
+#' Calls [xgboost::xgb.train()] from package \CRANpkg{xgboost} with `objective`
+#' set to `survival:aft` by default.
+#'
+#' @details
+#' This learner returns three prediction types:
+#' 1. `response`: the estimated survival time \eqn{T} for each test observation.
+#' 2. `lp`: a vector of linear predictors (relative risk scores), one per
+#' observation, estimated as \eqn{-log(T)}.
+#' Higher survival time denotes lower risk.
+#' 3. `crank`: same as `lp`.
+#'
+#' @template note_xgboost
+#'
+#' @section Initial parameter values:
+#' - `nrounds` is initialized to 1.
+#' - `nthread` is initialized to 1 to avoid conflicts with parallelization via \CRANpkg{future}.
+#' - `verbose` is initialized to 0.
+#' - `objective` is initialized to `survival:aft`.
```

Comment:
Ok I don't understand this. Why is there still the `objective` parameter when the learner sets it to aft?

Reply:
aft-nloglik is not documented

Reply:
Added the doc to `aft-nloglik`. Explained the other question in the below comment